### PR TITLE
kvutils: Remove the DAR upload parameters from the runner.

### PR DESCRIPTION
**Pull request by SamirTalwar-DA**
No production ledger needs to upload DARs on participant startup; this
feature is primarily for users of Sandbox. The feature never worked in
the case of multiple participants and was only ever used in testing.

This also removes the associated functionality from Sandbox-on-X as they
share a configuration object. Hopefully this won't be an issue.

Some tests were using this feature, so I have updated them to upload the
DARs through the PackageManagementService instead.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/MinVersionTest.scala [hunk 1]
```diff
@@ -96,27 +96,29 @@ final class MinVersionTest
...
-                          Seq(
-                            RecordField("issuer", Some(Value().withParty(party))),
-                            RecordField("owner", Some(Value().withParty(party))),
-                            RecordField("currency", Some(Value().withText("EUR"))),
-                            RecordField("amount", Some(Value().withNumeric("10.0"))),
-                            RecordField("observers", Some(Value().withList(List()))),
-                          ),
-                        )
-                      ),
+                  Command.of(
```
Comment (cocreature) [inline daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/MinVersionTest.scala:99]: seems like an unnecessary change which doesn’t really simplify things.

File: daml-script/test/src/test-utils/com/daml/lf/engine/script/test/MultiParticipantFixture.scala [hunk 2]
```diff
@@ -73,25 +82,41 @@ trait MultiParticipantFixture
...
             .createDefault(())
             .copy(
-              participants = Seq(participant1, participant2),
-              archiveFiles = Seq(darFile),
+              participants = Seq(participant1, participant2)
+            )
+        )
+        participant1Port = readPortfile(participant1Portfile)
+        participant2Port = readPortfile(participant2Portfile)
+        channel = {
```
Comment (nicu-da) [inline daml-script/test/src/test-utils/com/daml/lf/engine/script/test/MultiParticipantFixture.scala:98]: Do we clean this up after uploading packages?

File: ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/LedgerFactory.scala [hunk 3]
```diff
@@ -39,11 +39,8 @@ trait LedgerFactory[ExtraConfig] {
 }
 
 trait ReadWriteServiceFactory {
-
   def readService(): ReadService
 
-  def writePackagesService(): WritePackagesService
```
Comment (nicu-da) [inline ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/LedgerFactory.scala:45]: I cry tears of joy

File: ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/ConfigProvider.scala [hunk 4]
```diff
@@ -54,7 +54,7 @@ trait ConfigProvider[ExtraConfig] {
   ): ApiServerConfig =
     ApiServerConfig(
       participantId = participantConfig.participantId,
-      archiveFiles = config.archiveFiles.map(_.toFile).toList,
+      archiveFiles = Nil,
```
Comment (hubert-da) [inline ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/ConfigProvider.scala:57]: Do we need this one?

Comment (nicu-da) [inline ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/ConfigProvider.scala:57]: Is this still used?

File: daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/MinVersionTest.scala (hunk 1 above)
Comment (SamirTalwar-DA) [reply daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/MinVersionTest.scala:99]: Sorry, went a bit overboard; it was just to make IntelliJ stop complaining. I shall revert it.

File: daml-script/test/src/test-utils/com/daml/lf/engine/script/test/MultiParticipantFixture.scala (hunk 2 above)
Comment (SamirTalwar-DA) [reply daml-script/test/src/test-utils/com/daml/lf/engine/script/test/MultiParticipantFixture.scala:98]: Oh, good point, thanks!

File: ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/ConfigProvider.scala (hunk 4 above)
Comment (SamirTalwar-DA) [reply ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/ConfigProvider.scala:57]: Yes, by Sandbox.

**Review by hubert-da — APPROVED** [review]
LGTM 👍🏻 (just the previous comments)

File: daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/MinVersionTest.scala (hunk 1 above)
Comment (hubert-da) [reply daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/MinVersionTest.scala:99]: There was an issue with using a constructor for ScalaPB generated classes. I believe the same issue exists for `with`. @SamirTalwar-DA, do you recall it?

File: ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/ConfigProvider.scala (hunk 4 above)
Comment (hubert-da) [reply ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/ConfigProvider.scala:57]: Answered above.

Comment (SamirTalwar-DA) [reply ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/ConfigProvider.scala:57]: Yes, the property is still used by Sandbox.

File: daml-script/test/src/test-utils/com/daml/lf/engine/script/test/MultiParticipantFixture.scala (hunk 2 above)
Comment (SamirTalwar-DA) [reply daml-script/test/src/test-utils/com/daml/lf/engine/script/test/MultiParticipantFixture.scala]: Fixed.

File: daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/MinVersionTest.scala (hunk 1 above)
Comment (SamirTalwar-DA) [reply daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/MinVersionTest.scala]: No, not from memory, and this clearly worked before, so I have reverted it.

**Review by nicu-da — APPROVED** [review]

**Review by cocreature — APPROVED** [review]
:shipit:

**Review by tudor-da — APPROVED** [review]
nice simplfiication. 👍 from me provided that there is no impact on Sandbox users
